### ISSUE TITLE
wp-cli not handling locales that are not updated

### ISSUE THREAD

**Issue by dbrosy** [wp-cli/wp-cli#5439]
`Current version of WordPress: 5.5.1`
`Current Version of PHP: 7.4.9`

### Issue 1
When deploying a WordPress core that is not the latest wp-cli gives error.

Steps to reproduce

`wp core download  --path="/var/www/html" --locale="en_NZ" --force --quiet`

This currently returns

`Error: The requested locale (en_NZ) was not found.`

This is because "en_NZ" is currently at 5.4.2. 

The --quiet flag should hide this error and set the exit code to 1 (error) so it can be handled in script correctly or add another flag for suppressing all messages

### Issue 2

When deploying a language locale that is not up to current version

Steps

first install core

`wp core download  --path="/var/www/html" --locale="en_US" --force --quiet`

then

`wp language core install en_NZ`

this returns following:

`wp language core install en_NZ
PHP Warning:  Declaration of WP_CLI\LanguagePackUpgrader::download_package($package, $check_signatures = false) should be compatible with WP_Upgrader::download_package($package, $check_signatures = false, $hook_extra = Array) in phar:///usr/local/bin/wp-cli.phar/vendor/wp-cli/language-command/src/WP_CLI/LanguagePackUpgrader.php on line 51
Warning: Declaration of WP_CLI\LanguagePackUpgrader::download_package($package, $check_signatures = false) should be compatible with WP_Upgrader::download_package($package, $check_signatures = false, $hook_extra = Array) in phar:///usr/local/bin/wp-cli.phar/vendor/wp-cli/language-command/src/WP_CLI/LanguagePackUpgrader.php on line 51
Using cached file '/var/www/.wp-cli/cache/translation/core-default-5.4.2-en_NZ-1585693773.zip'...
Unpacking the update...
Installing the latest version...
Removing the old version of the translation...
Translation updated successfully.
Language 'en_NZ' installed.
Success: Installed 1 of 1 `languages.`

with the --quiet flag:

`wp language core install en_NZ --quiet
PHP Warning:  Declaration of WP_CLI\LanguagePackUpgrader::download_package($package, $check_signatures = false) should be compatible with WP_Upgrader::download_package($package, $check_signatures = false, $hook_extra = Array) in phar:///usr/local/bin/wp-cli.phar/vendor/wp-cli/language-command/src/WP_CLI/LanguagePackUpgrader.php on line 51
Warning: Declaration of WP_CLI\LanguagePackUpgrader::download_package($package, $check_signatures = false) should be compatible with WP_Upgrader::download_package($package, $check_signatures = false, $hook_extra = Array) in phar:///usr/local/bin/wp-cli.phar/vendor/wp-cli/language-command/src/WP_CLI/LanguagePackUpgrader.php on line 51`

After either of the above commands are run the locale is installed but with reports errors as above.

### Conclusion:

Error handling (suppression) is not possible with the --quiet flag and that wp-cli doesn't handle locales if they are not available for the latest version

**Comment by schlessera** [comment]
The `--quiet` flag "suppresses informational messages", not errors.

If you want to execute WP-CLI but not output any error messages, you should capture and redirect `STDERR`.

For example, the following would redirect `STDERR` (= file descriptor `2`) into a log file instead of printing it to the terminal:
```
wp some command 2>>error.log
```

For more on I/O redirection, see https://tldp.org/LDP/abs/html/io-redirection.html

**Comment by schlessera** [comment]
The fact that the language is said to be installed even though it doesn't exist seems to be a bug.

**Comment by chriscroome** [comment]
I'm seeing the same messages with English (UK):

```
PHP Warning:  Declaration of WP_CLI\LanguagePackUpgrader::download_package($package, $check_signatures = false) should be compatible with WP_Upgrader::download_package($package, $check_signatures = false, $hook_extra = Array) in phar:///usr/local/bin/wp/vendor/wp-cli/language-command/src/WP_CLI/LanguagePackUpgrader.php on line 112
Warning: Declaration of WP_CLI\LanguagePackUpgrader::download_package($package, $check_signatures = false) should be compatible with WP_Upgrader::download_package($package, $check_signatures = false, $hook_extra = Array) in phar:///usr/local/bin/wp/vendor/wp-cli/language-command/src/WP_CLI/LanguagePackUpgrader.php on line 112
```

**Comment by chalda-pnuzig** [comment]
Is there any update? 

```
# wp cli version
WP-CLI 2.4.0

# wp language plugin update --all
Updating translation for [...]...
PHP Warning:  Declaration of WP_CLI\LanguagePackUpgrader::download_package($package, $check_signatures = false) should be compatible with WP_Upgrader::download_package($package, $check_signatures = false, $hook_extra = Array) in phar:///usr/local/bin/wp/vendor/wp-cli/language-command/src/WP_CLI/LanguagePackUpgrader.php on line 51
Warning: Declaration of WP_CLI\LanguagePackUpgrader::download_package($package, $check_signatures = false) should be compatible with WP_Upgrader::download_package($package, $check_signatures = false, $hook_extra = Array) in phar:///usr/local/bin/wp/vendor/wp-cli/language-command/src/WP_CLI/LanguagePackUpgrader.php on line 51
Downloading translation from [...]...
Unpacking the update...
Installing the latest version...
Removing the old version of the translation...
Translation updated successfully.
Success: Updated 1/1 translation.
`

**Comment by Ippey** [comment]
This issue seems to be resolved on `2.5`, thanks to the below PR.
This could be closed, I think.

https://github.com/wp-cli/language-command/pull/100
